app/components/RoleTable.tsx

```diff
@@ -1,13 +1,65 @@
 'use client';
 
 import React from 'react';
-import { TableContainer, Table, TableRow, TableHeader, TableCell, Link, SignerInfo, RequiredSigners, TotalSigners } from '../styles/components';
+import { 
+    TableContainer, 
+    Table, 
+    TableRow, 
+    TableHeader, 
+    TableCell, 
+    Link, 
+    SignerInfo, 
+    RequiredSigners, 
+    TotalSigners,
+    SignerName,
+    OnlineKey,
+    ThresholdInfo,
+    SignersList
+} from '../styles/components';
 import { RoleInfo } from '../utils/types';
 
 interface RoleTableProps {
     roles: RoleInfo[];
 }
 
+// Helper function to format the keyid for display (show as username)
+const formatKeyId = (keyid: string): string => {
+    // If it's already a username format with @, just return it
+    if (keyid.startsWith('@')) {
+        return keyid;
+    }
+    
+    // If it's an online key, return as is
+    if (keyid.toLowerCase() === 'online key') {
+        return keyid;
+    }
+    
+    // Check if the keyid looks like a hex string and truncate it for display
+    if (keyid.match(/^[0-9a-f]+$/i)) {
+        return `@${keyid.substring(0, 8)}`;
+    }
+    
+    // Otherwise, add @ to make it look like a username
+    return `@${keyid}`;
+};
+
+// Helper to determine if a key is an "online key"
+const isOnlineKey = (keyid: string): boolean => {
+    // If the key ID is already labeled as an online key
+    if (keyid.toLowerCase() === 'online key') {
+        return true;
+    }
+    
+    // Check for common patterns in online key IDs
+    if (keyid.toLowerCase().includes('online') || 
+        keyid === '0c87432c' ||
+        keyid === '5e3a4021') {
+        return true;
+    }
+    
+    return false;
+};
+
```

Comment:
Please consider updating formatKeyId and isOnlineKey as suggested in https://github.com/DeshDeepakKant/TUF-Metadata-Visualizer/pull/8#discussion_r2030808995. You can do it as part of #5.